modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -584,6 +585,7 @@ void CvCapture_FFMPEG::init()
     va_type = cv::VIDEO_ACCELERATION_NONE;  // TODO OpenCV 5.0: change to _ANY?
     hw_device = -1;
     use_opencl = 0;
+    extraDataIdx = 1;
```

Comment:
Some codecs doesn't use/have extra data. So we should not force that.

Reply:
Do you mean that
cap.get(CAP_PROP_CODEC_EXTRADATA_INDEX) > 0
is there to indicate that extra data is supported and that it exists not just that it is supported by the video backend?

Reply:
Where should we define the index for the extra data index?

Reply:
Currently we have `cap.get(CAP_PROP_CODEC_EXTRADATA_INDEX) > 0` iff backend supports this property
- it is still true if there is no support for used codec
- it is still true if this data is empty

I'm OK with that for now.